PianoFromAbove/GameState.cpp

```diff
@@ -2063,41 +1948,144 @@ void MainScreen::RenderStatusLine(int line, const char* left, const char* format
 
 void MainScreen::RenderStatus(LPRECT prcStatus)
 {
-    // Time
+    // Playback statues
+    long long nps = 0;
+        for (int i = 0; i < m_dNPSNotes.size(); i++)
+            nps += std::get<1>(m_dNPSNotes[i]);
+    size_t state_size = 0;
+    for (auto note_state : m_vState)
+        state_size += note_state.size();
     Config& config = Config::GetConfig();
     VizSettings viz = config.GetVizSettings();
+    static ViewSettings& cView = config.GetViewSettings();
+    const PlaybackSettings& cPlayback = config.GetPlaybackSettings();
     const MIDI::MIDIInfo& mInfo = m_MIDI.GetInfo();
-    auto starttime = ((m_llStartTime >= 0) ? m_llStartTime : -m_llStartTime);
-
-    auto min = starttime / 60000000;
-    auto sec = (starttime % 60000000) / 1000000;
-    auto cs = (starttime % 1000000) / 100000;
-    auto tmin = mInfo.llTotalMicroSecs / 60000000;
-    auto tsec = (mInfo.llTotalMicroSecs % 60000000) / 1000000;
-    auto tcs = (mInfo.llTotalMicroSecs % 1000000) / 100000;
-    auto tempo = 60000000.0 / m_iMicroSecsPerBeat;
-
     int cur_line = 0;
     m_pRenderer->GetDrawList()->AddRectFilled(ImVec2(prcStatus->left, prcStatus->top), ImVec2(prcStatus->right, prcStatus->bottom), 0x80000000);
-
-    RenderStatusLine(cur_line++, "Time:", "%s%lld:%02d.%d / %lld:%02d.%d",
-        m_llStartTime >= 0 ? "" : "-",
-        min, sec, cs,
-        tmin, tsec, tcs);
-    RenderStatusLine(cur_line++, "Tempo:", "%.3lf bpm", tempo);
-
-    // Framerate
-    if (m_bShowFPS && !m_bDumpFrames)
-        RenderStatusLine(cur_line++, "FPS:", "%.1lf", m_dFPS);
-
-    // Nerd stats
-    if (viz.bNerdStats) {
-        long long nps = 0;
-        for (int i = 0; i < m_dNPSNotes.size(); i++)
-            nps += std::get<1>(m_dNPSNotes[i]);
-
-        RenderStatusLine(cur_line++, "NPS:", "%lld", nps);
-        RenderStatusLine(cur_line++, "Rendered:", "%llu", m_pRenderer->GetRenderedNotesCount());
+    RenderStatusLine(cur_line++,"PianoFromAbove VIZ", "v3.1.0");
+    RenderStatusLine(cur_line++,"Made by: happy_mimimix", "");
```

Comment:
PFA Viz wasn't made by you, even if you have a couple of your own changes in there.